### PR TITLE
Multi: Update dcrd and dcrwallet dependencies

### PR DESCRIPTION
**Pull request by matheusd**
This updates the dcrwallet and dcrd dependencies to their current (in-development) version.

Due to the strong interdependency between the various packages, it's not really possible to do partial migrations that still work and pass all tests: all packages need to be updated in pretty much one go.

The most significant change is the introduction of the `stdaddr` package in dcrd, which replaces uses of `dcrutil.Address` throughout the codebases.

Another change included in this PR is building and using a uniquely named `dcrd` binary for unit and integration tests (`dcrd-dcrlnd`) now that the upstream dcrd rpctest package supports specifying the dcrd binary to use. This avoids having to have the local dcrd binary replaced by a potentially in-development version when running the dcrlnd tests.

### CODE REVIEW

File: chainntnfs/dcrdnotify/dcrd.go [hunk 1]
```diff
@@ -5,16 +5,17 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline chainntnfs/dcrdnotify/dcrd.go:8]: may want to group this with the non-stdlib imports below

File: chainntnfs/dcrdnotify/dcrd_test.go [hunk 2]
```diff
@@ -5,13 +5,13 @@ package dcrdnotify
 import (
 	"bytes"
 	"context"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline chainntnfs/dcrdnotify/dcrd_test.go:8]: same

File: channeldb/migration_01_to_11/zpay32/invoice.go [hunk 3]
```diff
@@ -3,11 +3,12 @@ package zpay32
 import (
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline channeldb/migration_01_to_11/zpay32/invoice.go:6]: same

File: contractcourt/chain_watcher.go [hunk 4]
```diff
@@ -3,15 +3,16 @@ package contractcourt
 import (
 	"bytes"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline contractcourt/chain_watcher.go:6]: same

File: input/test_utils.go [hunk 5]
```diff
@@ -4,14 +4,16 @@ import (
 	"bytes"
 	"encoding/hex"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/sign"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline input/test_utils.go:8]: same

File: lnd.go [hunk 6]
```diff
@@ -8,6 +8,7 @@ import (
 	"context"
 	"crypto/tls"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline lnd.go:11]: same

File: lnwallet/chanvalidate/validate_test.go [hunk 7]
```diff
@@ -2,14 +2,15 @@ package chanvalidate
 
 import (
 	"bytes"
+	"github.com/decred/dcrd/txscript/v4/sign"
```
Comment (jrick) [inline lnwallet/chanvalidate/validate_test.go:5]: same

File: lnwallet/interface.go [hunk 8]
```diff
@@ -3,13 +3,14 @@ package lnwallet
 import (
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline lnwallet/interface.go:6]: same

File: lnwallet/remotedcrwallet/wallet.go [hunk 9]
```diff
@@ -4,23 +4,25 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline lnwallet/remotedcrwallet/wallet.go:7]: same

File: lnwallet/wallet.go [hunk 10]
```diff
@@ -5,18 +5,19 @@ import (
 	"crypto/sha256"
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline lnwallet/wallet.go:8]: same

File: mock.go [hunk 11]
```diff
@@ -3,18 +3,19 @@ package dcrlnd
 import (
 	"encoding/hex"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/sign"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline mock.go:7]: same

File: rpcserver.go [hunk 12]
```diff
@@ -7,6 +7,7 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline rpcserver.go:10]: same

File: zpay32/decode.go [hunk 13]
```diff
@@ -4,16 +4,17 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline zpay32/decode.go:7]: same

File: zpay32/encode.go [hunk 14]
```diff
@@ -4,10 +4,11 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline zpay32/encode.go:7]: same

File: zpay32/invoice.go [hunk 15]
```diff
@@ -3,11 +3,12 @@ package zpay32
 import (
 	"errors"
 	"fmt"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline zpay32/invoice.go:6]: same

File: zpay32/invoice_internal_test.go [hunk 16]
```diff
@@ -2,6 +2,7 @@ package zpay32
 
 import (
 	"encoding/binary"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
```
Comment (jrick) [inline zpay32/invoice_internal_test.go:5]: same

File: chainntnfs/dcrdnotify/dcrd.go (hunk 1 above)
Comment (matheusd) [reply chainntnfs/dcrdnotify/dcrd.go]: Thank you for the review! These were all added automatically via some scripts I wrote to ease the conversion process, that's why there were ungrouped.

File: chainscan/csdrivers/dcrwdriver.go [hunk 17]
```diff
@@ -5,10 +5,11 @@ import (
...
 	"time"
 
-	"decred.org/dcrwallet/errors"
-	"decred.org/dcrwallet/wallet"
+	"decred.org/dcrwallet/v2/errors"
+	"decred.org/dcrwallet/v2/wallet"
 	"github.com/decred/dcrd/chaincfg/chainhash"
-	"github.com/decred/dcrd/gcs/v2"
+	"github.com/decred/dcrd/gcs/v3"
+	gcs3 "github.com/decred/dcrd/gcs/v3"
```
Comment (matheusd) [inline chainscan/csdrivers/dcrwdriver.go:12]: Duplicated gcs pkg

**Pull request merged**